lib/spack/spack/test/package_sanity.py

```diff
@@ -56,6 +57,19 @@ def test_packages_are_pickleable():
             pickle.dumps(pkg)
 
 
+def test_packages_are_unparseable():
+    failed_to_unparse = list()
+    for name in spack.repo.all_package_names():
+        try:
+            ph.canonical_source(name, filter_multimethods=False)
+        except Exception:
+            failed_to_unparse.append(name)
+
+    if failed_to_unparse:
+        tty.msg('The following packages failed to unparse: ' +
+                ', '.join(failed_to_unparse))
```

Comment:
Is this unfinished code? I would assume we'd use an `assert` here in order to make sure this test fails?